front-end/src/pages/TeachersList/index.jsx

```diff
@@ -0,0 +1,54 @@
+/* eslint-disable react/jsx-indent */
+/* eslint-disable indent */
+import React, { useEffect, useState } from 'react';
+
+// Service
+import { teacherService } from 'services/teacherService';
+
+// Components
+import Card from 'components/TeachersList/Card';
+import Loader from 'components/common/Loader/Loader';
+
+// Styles
+import styles from './TeachersList.module.scss';
+
+const TeachersList = () => {
+  const [teachers, setTeachers] = useState([]);
+
+  const fetchTeachers = async () => {
+    try {
+      const data = await teacherService.getAllTeacher();
+      setTeachers(data);
+    } catch (e) {
+      console.log(e);
+    }
+  };
+
+  useEffect(() => {
+    fetchTeachers();
+  }, []);
+
+  return (
+    <div className={styles.wrapper}>
+      <h1>Teachers</h1>
+      {teachers.length ? (
+        <div className={styles.cardWrap}>
+          {teachers.length
+            ? teachers.map((teacher) => (
+                <Card
+                  key={teacher?._id}
+                  link={`/app/teachers/${teacher?._id}/overview`}
```

Comment:
You could pass teacher id and create link in Card component

Reply:
fix it